pkgs/applications/editors/vim/plugins/avante-nvim/default.nix

```diff
@@ -0,0 +1,74 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  openssl,
+  stdenv,
+  vimUtils,
+  darwin,
+}:
+
+let
+  version = "2024-09-15";
+
+  src = fetchFromGitHub {
+    owner = "yetone";
+    repo = "avante.nvim";
+    rev = "f9520c4fdfed08e9cc609d6cd319b358e4ea33a5";
+    hash = "sha256-8zTDGPnhNI2rQA0uJc8gQRj4JCyg+IkO/D3oHYy4f9U=";
+  };
```

Comment:
I would recommend to follow the same approach as in #341927 and use the upstream releases (0.0.2 currently).
Then, you will be able to add `passthru.updateScript = nix-update-script { };` which should make our lives easier.

Reply:
I don't like moving to upstream releases because
- 0.0.2 is broken (see comments above)
- I don't want to impose a release schedule on upstream
- Almost all other plugins specify a commit, not a release tag

Regarding the update script, i did some limited testing, it seemed to not update Cargo.lock. Maybe user error.
The only vim plugin that has updateScript is https://github.com/NixOS/nixpkgs/blob/f9e21602e2b6954ee32056c9525dc150babf5dc5/pkgs/applications/editors/vim/plugins/vim-clap/default.nix#L70
hmm, is it used just to more easily populate outputHashes?

Reply:
> I don't like moving to upstream releases because
> 
>     * 0.0.2 is broken (see comments above)
> 
>     * I don't want to impose a release schedule on upstream
> 
>     * Almost all other plugins specify a commit, not a release tag

Fair enough !